Dockerfile

```diff
@@ -0,0 +1,85 @@
+FROM python:3.8-slim-buster as base
+
+ENV PYTHONFAULTHANDLER=1 \
+  PYTHONUNBUFFERED=1 \
+  PYTHONHASHSEED=random \
+  PIP_NO_CACHE_DIR=off \
+  PIP_DISABLE_PIP_VERSION_CHECK=on \
+  PIP_DEFAULT_TIMEOUT=100 \
+  POETRY_VERSION=1.0.1 \
+  TRELLO_BOARD=$TRELLO_BOARD \
+  TRELLO_KEY=$TRELLO_KEY \
+  TRELLO_TOKEN=$TRELLO_TOKEN
+
+# System deps:
+RUN pip install "poetry==$POETRY_VERSION"
+
+# Copy only requirements to cache them in docker layer
+WORKDIR /code
+COPY poetry.lock pyproject.toml /code/
+
+# Project initialization:
+RUN poetry config virtualenvs.create false \
+  && poetry install --no-interaction --no-ansi
+
+#------------#
+# PRODUCTION #
+#------------#
+
+# Setting up the production image
+FROM base as production
+
```

Comment:
There's a bit more you could do to make production and development different.
E.g. `ENV FLASK_ENV=production`

And you could move poetry install into the stages, so that production can skip pytest etc by using the `--no-dev` option